documentation/doxygen/makerootfile.C

```diff
@@ -0,0 +1,58 @@
+/// Generates the root file output of the macro MacroName
+
+
+
+void makerootfile(const char *MacroName, const char *IN, const char *OutDir, bool cp, bool py)
+{
+
+   // Execute the macro as a C++ one or a Python one.
+   if (!py) gROOT->ProcessLine(Form(".x %s",MacroName));
+   else     gROOT->ProcessLine(Form("TPython::ExecScript(\"%s\");",MacroName));
+
+   // If needed, copy the macro in the documentation directory.
+   if (cp) {
+      TString MN = MacroName;
+      Int_t i = MN.Index("(");
+      Int_t l = MN.Length();
+      if (i>0) MN.Remove(i, l);
+      gSystem->Exec(TString::Format("cp %s %s/macros", MN.Data(), OutDir));
+   }
+
+   // Iterate over all the pads generated by the macro and produce the corresponding pictures.
+   TIter iCanvas(gROOT->GetListOfCanvases());
+   TCanvas* canvas = 0;
+   int ImageNum = 0;
+
+   auto f = new TFile( TString::Format("%s/html/%s.root",OutDir,IN),"RECREATE");
+
+   int cw, ch;
+   while ((canvas = (TCanvas*) iCanvas())) {
+      ImageNum++;
+      canvas->SetName(TString::Format("pict%d_%s",ImageNum,IN));
+      cw = canvas->GetWindowWidth();
+      ch = canvas->GetWindowHeight();
+      canvas->Write();
+   }
+
+   f->Close();
+
+   // Build the html file inlining the JS pictures
+   int i;
+   FILE *fh = fopen(TString::Format("%s/macros/%s.html",OutDir,IN), "w");
+   fprintf(fh,"<script src=\"https://root.cern.ch/js/dev/scripts/JSRootCore.js\" type=\"text/javascript\"></script>\n");
```

Comment:
This can be a problem.
If there are many JSROOT images on the same HTML page, JSRootCore.js will be loaded many times.
One could use following JS code:

```
if (!JSROOT) {
   var element = document.createElement("script");
   element.setAttribute("type", "text/javascript");
   element.setAttribute("src", "https://root.cern.ch/js/dev/scripts/JSRootCore.js");
   document.getElementsByTagName("head")[0].appendChild(element);
}
```

Reply:
Yes I was wondering ... I am now looking at cases with several images

Reply:
Another solution - just check if JSRootCore.js script already there in HTML file. This requires to read actual file content before writing new data into it.

Reply:
Hello Sergey,

In case of several pictures, the code generated is the following right now:

```
<script type='text/javascript'>
JSROOT.OpenFile("./ContourList.root", function(file) {
   file.ReadObject("pict1_ContourList", function(obj) {JSROOT.draw("draw_pict1_ContourList", obj, "");});
   file.ReadObject("pict2_ContourList", function(obj) {JSROOT.draw("draw_pict2_ContourList", obj, "");});
});
</script>
<center>
<div id="draw_pict1_ContourList" style="width:600px; height:600px"></div>
<div id="draw_pict2_ContourList" style="width:600px; height:600px"></div>
</center>
```

Reply:
Yes, code is ok.
Reading of both objects (pict1_ContourList and pict2_ContourList) will be done in parallel.
In such case one should avoid use of sub-directories in TFile.

Reply:
They are flat files. No sub-folders.